StartBonusActivation/handler.ts

```diff
@@ -185,6 +187,14 @@ const getLastValidDSU = (
     }
   );
 
+/**
+ * Generate a random bonus code. The operation may fail, so it has a retry mechanism
```

Comment:
actually, this cannot fail. it's ok to just wrap it in `tryCatch` (without retries)

Reply:
the script has actually chances to fail: https://github.com/pagopa/io-functions-bonus/blob/7fb8cd62f9fb0e74dbc92d4e6fa0da15fc9a77bd/utils/bonusCode.ts#L33

Being randomly generated, it may result in non-deterministic failures. Retries are meant to reduce this risk, although being maybe an over-engineering

Reply:
yes, that exception won't never throw, it's just a guard to extract the right side of the Either (that could have been a simple cast to BonusCode) so there's no need to retry here